src/util.h

```diff
@@ -117,6 +117,20 @@ namespace rhost {
             }
         };
 
+        struct blob {
+            void* raw_data;
```

Comment:
For a view, this should probably be `const void`, since we don't actually mutate the data anywhere, and sender might have an immutable buffer to send from (say, `const std::string`).